### PR TITLE
Only show focus rings on buttons for keyboard navigation

### PR DESCRIPTION
**Pull request by markdalgleish**
See changeset for details.

### CODE REVIEW

File: lib/components/private/hideFocusRings/HideFocusRingsRoot.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export const HideFocusRingsRoot = ({ children }: HideFocusRingsRootProps) => {
+  const [hideFocusRings, setHideFocusRings] = useState(false);
+
+  useEffect(() => {
+    const eventName = hideFocusRings ? 'keydown' : 'mousemove';
+    const toggleFocusRings = () => setHideFocusRings(x => !x);
+
+    window.addEventListener(eventName, toggleFocusRings);
```
Comment (mattsjones) [inline lib/components/private/hideFocusRings/HideFocusRingsRoot.tsx]: 😍

**Comment by markdalgleish** [comment]
Updated this so that we bail out of wiring up the focus ring logic when nested within a `BraidTestProvider`. Worth a re-review.

File: setupTests.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { format } from 'util';
+
+const error = global.console.error;
+
+global.console.error = (message: any, ...restArgs: any[]) => {
```
Comment (mattsjones) [inline setupTests.ts]: Nice

File: .changeset/stale-bags-eat.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+'braid-design-system': patch
```
Comment (mattsjones) [inline .changeset/stale-bags-eat.md]: Feel like `minor` would be more appropriate. But won't matter either way.

Comment (markdalgleish) [reply .changeset/stale-bags-eat.md]: Changed.

**Pull request merged**